File: app/src/main/java/com/sdwfqin/quickseed/ui/components/PictureUploadActivity.java
package com.sdwfqin.quickseed.ui.components;

import android.content.Intent;
import android.net.Uri;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.sdwfqin.quickseed.base.ArouterConstants;
import com.sdwfqin.quickseed.base.SampleBaseActivity;
import com.sdwfqin.quickseed.databinding.ActivityPictureUploadBinding;
import com.sdwfqin.quickseed.model.PictureModel;
import com.sdwfqin.quickseed.utils.picture.PictureSelectUtils;
import com.sdwfqin.widget.pictureupload.PictureUploadCallback;
import com.sdwfqin.widget.pictureupload.PictureUploadView;
import com.zhihu.matisse.Matisse;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

/**
 * 描述：九宫格上传图片
 *
 * @author zhangqin
 * @date 2018/5/31
 */
@Route(path = ArouterConstants.COMPONENTS_PICTUREUPLOAD)
public class PictureUploadActivity extends SampleBaseActivity<ActivityPictureUploadBinding> {

    public static final int RESULT_PHOTO_SELECT = 101;

//    PictureUploadView<PictureModel> mPic;

    @Override
    protected ActivityPictureUploadBinding getViewBinding() {
        return ActivityPictureUploadBinding.inflate(getLayoutInflater());
    }

    @Override
    protected void initEventAndData() {
        mTopBar.setTitle("九宫格上传图片");
        mTopBar.addLeftBackImageButton()
                .setOnClickListener(v -> finish());

//        mPic = mBinding.pic;

        mBinding.pic.setMaxColumn(3);
        mBinding.pic.setPicUploadCallback(new PictureUploadCallback<PictureModel>() {
            @Override
            public void click(int position, PictureModel pictureModel, List<PictureModel> list) {

            }

            @Override
            public void remove(int position, List list) {

            }

            @Override
            public void onAddPic(int maxPic, List list) {
                PictureSelectUtils.SelectSystemPhoto(mContext, RESULT_PHOTO_SELECT, maxPic);
            }
        });
    }

    @Override
    protected void initClickListener() {

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (resultCode == RESULT_OK) {
            switch (requestCode) {
                case RESULT_PHOTO_SELECT:
                    List<PictureModel> models = new ArrayList<>();
                    List<String> selectList = Matisse.obtainPathResult(data);
                    for (int i = 0; i < selectList.size(); i++) {
                        models.add(new PictureModel(selectList.get(i)));
                    }
                    mBinding.pic.setAddData(models);

                    // 刷新相册图片
                    if (selectList.size() == 1) {
                        String saveAs = selectList.get(0);
                        Uri contentUri = Uri.fromFile(new File(saveAs));
                        Intent mediaScanIntent = new Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE, contentUri);
                        sendBroadcast(mediaScanIntent);
                    }
                    break;
                default:
            }
        }
    }
}

File: app/src/main/java/io/github/sdwfqin/quickseed/ui/components/PictureUploadActivity.kt
package io.github.sdwfqin.quickseed.ui.components

import android.content.Intent
import android.view.View
import com.alibaba.android.arouter.facade.annotation.Route
import com.zhihu.matisse.Matisse
import io.github.sdwfqin.quickseed.constants.ArouterConstants
import io.github.sdwfqin.quickseed.data.bean.PictureModel
import io.github.sdwfqin.quickseed.databinding.ActivityPictureUploadBinding
import io.github.sdwfqin.samplecommonlibrary.base.SampleBaseActivity
import io.github.sdwfqin.samplecommonlibrary.utils.picture.PictureSelectUtils
import io.github.sdwfqin.widget.pictureupload.PictureUploadCallback
import java.util.*

/**
 * 描述：九宫格上传图片
 *
 * @author zhangqin
 * @date 2018/5/31
 */
@Route(path = ArouterConstants.COMPONENTS_PICTUREUPLOAD)
class PictureUploadActivity : SampleBaseActivity<ActivityPictureUploadBinding>() {

    override fun getViewBinding(): ActivityPictureUploadBinding {
        return ActivityPictureUploadBinding.inflate(
            layoutInflater
        )
    }

    override fun initEventAndData() {
        mTopBar.setTitle("九宫格上传图片")
        mTopBar.addLeftBackImageButton()
            .setOnClickListener { v: View? -> finish() }
        mBinding.pic.setMaxColumn(3)
        mBinding.pic.setMaxSize(12)
        mBinding.pic.setPicUploadCallback(object : PictureUploadCallback<PictureModel> {
            override fun click(
                position: Int,
                item: PictureModel,
                list: List<PictureModel>
            ) {

            }

            override fun remove(position: Int, list: List<PictureModel>): Boolean {
                return true
            }

            override fun onAddPic(maxPic: Int, list: List<PictureModel>) {
                PictureSelectUtils.SelectSystemPhoto(mContext, RESULT_PHOTO_SELECT, maxPic)
            }
        })
    }

    override fun initClickListener() {}

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == RESULT_OK) {
            when (requestCode) {
                RESULT_PHOTO_SELECT -> {
                    val models: MutableList<PictureModel> = ArrayList()
                    val selectList = Matisse.obtainResult(data)
                    var i = 0
                    while (i < selectList.size) {
                        models.add(PictureModel(selectList[i]))
                        i++
                    }
                    mBinding!!.pic.addData(models)

                    // 刷新相册图片
                    if (selectList.size == 1) {
                        val contentUri = selectList[0]
                        val mediaScanIntent =
                            Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE, contentUri)
                        sendBroadcast(mediaScanIntent)
                    }
                }
                else -> {
                }
            }
        }
    }

    companion object {
        const val RESULT_PHOTO_SELECT = 101
    }
}